### PR TITLE
MGDOBR-1320: Add Processor Worker for dependency management

### PR DESCRIPTION
**Pull request by manstis**
[https://issues.redhat.com/browse/MGDOBR-1320](https://issues.redhat.com/browse/MGDOBR-1320)

## PR owner checklist

Please make sure that your PR meets the following requirements:

- [x] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config).  
  *Run `mvn clean verify -DskipTests` so that imports are correctly set.*
- [x] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-XXX] - $clear_explanation_of_what_you_did"
- [x] The layers in the `kustomize` folder have been updated accordingly.
- [x] All new functionality is tested
- [x] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

## PR reviewer(s) checklist

- [ ] Check if code and Github action workflows have been modified and if the modification is safe
- [ ] If the modification is safe, add the `safe to test` label

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

- <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

- <b>Rebase the pull request</b>  
  Comment with: `/rebase`.
- <b>Deploy BOT</b>
  Comment with `/deploy <dev,stable> when the PR has been merged to deploy to a target environment.

</details>

### CODE REVIEW

File: manager-parent/manager-v2/src/main/java/com/redhat/service/smartevents/manager/v2/workers/resources/ProcessorWorker.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+import com.redhat.service.smartevents.manager.v2.persistence.dao.ManagedResourceV2DAO;
+import com.redhat.service.smartevents.manager.v2.persistence.dao.ProcessorDAO;
+import com.redhat.service.smartevents.manager.v2.persistence.models.Processor;
+
+@ApplicationScoped
+public class ProcessorWorker extends AbstractWorker<Processor> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ProcessorWorker.class);
+
+    private static final String PROCESSOR_WORKER_CLASSNAME = Processor.class.getName();
```
Comment (winklerm) [inline manager-parent/manager-v2/src/main/java/com/redhat/service/smartevents/manager/v2/workers/resources/ProcessorWorker.java:22]: One nitpick - the `PROCESSOR_WORKER_CLASSNAME` name makes me expect its value to be `ProcessorWorker.class.getName()`. 

I have checked it is actually consistent with `BridgeWorker.BRIDGE_WORKER_CLASSNAME` and it also makes sense this way, it is just not obvious. I am totally fine with keeping it as it is, just maybe a little food for thought for future refactoring?

Wdyt?

**Review by winklerm — APPROVED** [review]
LGTM - just one nitpicking comment which is definitely not a blocker for merging.

File: manager-parent/manager-v2/src/main/java/com/redhat/service/smartevents/manager/v2/workers/resources/ProcessorWorker.java (hunk 1 above)
Comment (manstis) [reply manager-parent/manager-v2/src/main/java/com/redhat/service/smartevents/manager/v2/workers/resources/ProcessorWorker.java:22]: Hi @winklerm. The `Worker` acts on certain `ManagedResources`.

We find a `Worker` that can handle `Work` for a `ManagedResource` [here](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/blob/main/manager-parent/manager-v2/src/main/java/com/redhat/service/smartevents/manager/v2/workers/quartz/WorkJob.java#L37). The `Work` is constructed from the `ManagedResource` [here](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/blob/main/manager-parent/manager-v2/src/main/java/com/redhat/service/smartevents/manager/v2/workers/quartz/QuartzWorkManagerImpl.java#L59) using [this](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/blob/main/manager-parent/manager-core/src/main/java/com/redhat/service/smartevents/manager/core/workers/Work.java#L50). So.. the class name of the `ManagedResource` is used in the `accept` method on the `Worker` that uses the constant here.

Does this make more sense?

I however agree with your comment and will change both in due course (once this PR is merged).

**Review by r00ta — APPROVED** [review]

File: manager-parent/manager-v2/src/main/java/com/redhat/service/smartevents/manager/v2/workers/resources/ProcessorWorker.java (hunk 1 above)
Comment (winklerm) [reply manager-parent/manager-v2/src/main/java/com/redhat/service/smartevents/manager/v2/workers/resources/ProcessorWorker.java:22]: Yes, it totally makes sense, thanks! 

My only concern (and indeed just a nitpick) was that the name makes you expect something slightly different from what is stored there (which is probably more apparent when someone not familiar with the code is reading it). But overall, it definitely makes sense, so it is just about improving further the readability.

**Pull request merged**